components/cli/pkg/commands/build_test.go

```diff
@@ -0,0 +1,193 @@
+}
+
+func TestExecuteTempBalFile(t *testing.T) {
+	mockCli := test.NewMockCli()
+	currentDir, _ := mockCli.FileSystem().CurrentDir()
+	mockBalExecutor := &test.MockBalExecutor{
+		CurrentDir: currentDir,
+	}
+	defer func() { os.RemoveAll(currentDir) }()
+	executableBal, _ := os.Create(filepath.Join(currentDir, "executable.bal"))
+	err := executeTempBalFile(mockBalExecutor, executableBal.Name(), []byte("test"))
+	if err != nil {
+		t.Errorf("Failed to run executable bal file: %v", err)
+	}
+	var artifacts []string
+	filepath.Walk(currentDir, func(path string, info os.FileInfo, err error) error {
+		artifacts = append(artifacts, path)
+		return nil
+	})
+	got := artifacts
+	expected := []string{currentDir, currentDir + "/metadata.json"}
+	if diff := cmp.Diff(expected, got); diff != "" {
+		t.Errorf("Write (-want, +got)\n%v", diff)
+	}
+}
+
+func TestGenerateMetaData(t *testing.T) {
+	mockCli := test.NewMockCli()
+	projectDir, _ := mockCli.FileSystem().CurrentDir()
+	targetDir := filepath.Join(projectDir, "target")
+	os.MkdirAll(filepath.Join(targetDir, "cellery"), os.ModePerm)
+	defer func() { os.RemoveAll(projectDir) }()
+	cellImage := &image2.CellImage{
+		Registry:     "registry.hub.cellery.io",
+		Organization: "myorg",
+		ImageName:    "aaa",
+		ImageVersion: "1.0.0",
+	}
+	mockMetaData, err := ioutil.ReadFile(filepath.Join("testdata", "metadata.json"))
+	if err != nil {
+		t.Errorf("Failed to read metadatajson.golden: %v", err)
+	}
+	mokcMetaDataJson, err := os.Create(filepath.Join(targetDir, "cellery", "metadata.json"))
+	if err != nil {
+		t.Errorf("Failed to create mock metadatajson: %v", err)
+	}
+	err = ioutil.WriteFile(mokcMetaDataJson.Name(), []byte(mockMetaData), 0644)
+	if err != nil {
+		t.Errorf("Failed to write to mock metadatajson: %v", err)
+	}
+	mockYamlData, err := ioutil.ReadFile(filepath.Join("testdata", "hello.yaml"))
+	if err != nil {
+		t.Errorf("Failed to read yaml.golden: %v", err)
+	}
+	mockYamlFile, err := os.Create(filepath.Join(targetDir, "cellery", "aaa.yaml"))
+	if err != nil {
+		t.Errorf("Failed to create mock yaml file: %v", err)
+	}
+	err = ioutil.WriteFile(mockYamlFile.Name(), []byte(mockYamlData), 0644)
+	if err != nil {
+		t.Errorf("Failed to write to mock yaml file: %v", err)
+	}
+	err = generateMetaData(cellImage, projectDir)
+	if err != nil {
+		t.Errorf("Failed to generate metadata: %v", err)
+	}
+	metaData, err := ioutil.ReadFile(mokcMetaDataJson.Name())
+	if err != nil {
+		t.Errorf("Failed to read temp executable bal file: %v", err)
+	}
+	got := &image2.MetaData{}
+	if err = json.Unmarshal(metaData, got); err != nil {
+		t.Errorf("error unmarshalling metadata json: %v", err)
+	}
+	if diff := cmp.Diff("myorg", got.Organization); diff != "" {
+		t.Errorf("Write (-want, +got)\n%v", diff)
+	}
+	if diff := cmp.Diff("aaa", got.Name); diff != "" {
+		t.Errorf("Write (-want, +got)\n%v", diff)
+	}
+	if diff := cmp.Diff("1.0.0", got.Version); diff != "" {
+		t.Errorf("Write (-want, +got)\n%v", diff)
+	}
+}
```

Comment:
Shall we add a test for RunBuild method.

Reply:
Will add.